### PR TITLE
Make latency prediction server multi threaded compatable

### PR DESCRIPTION
**Pull request by kaushikmitr**
This pull request introduces improvements to the prediction server's deployment configuration and updates the stress test parameters for bulk predictions. The main changes are focused on making the server more configurable and increasing the robustness of performance testing.

**Deployment configuration:**

* Added the `UVICORN_WORKERS` environment variable to the `Dockerfile-prediction`, allowing the number of Uvicorn worker processes to be configured via an environment variable. The default is set to 8.
* Modified the container startup command to use a shell and pass the `--workers` flag to Uvicorn, enabling dynamic worker configuration based on the environment variable.

**Testing improvements:**

* Increased the batch sizes used in the bulk prediction stress test from `[5, 10, 25]` to `[25, 50, 100]` in `test_dual_server_client.py`, making the stress test more demanding and representative of higher load scenarios.

### CODE REVIEW

**Comment by kaushikmitr** [comment]
@ahg-g @Gregory-Pereira @BenjaminBraunDev results are good by allowing multi threads. We still need to implement throttling though.


Previously i had 1 Uvicorn worker (default) with XGBoost using all available CPUs per prediction (OpenMP default). So single prediction at a time (due to the lock), using all cores for that one prediction via XGBoost's internal parallelism.

Below I have 28 predictions concurrently with each prediction using only 1 thread. The latter seems to give better throughput


### **Performance Test Results**

**Setup Configuration**

| Parameter | Value |
| :--- | :--- |
| **Machine** |  GCE e2-highcpu-32 — 32 vCPUs, 32 GB RAM (28 CPUs allocated to prediction server)  |
| **Prediction server replicas** | 1 |
| **`UVICORN_WORKERS`** | 28 |
| **`OMP_NUM_THREADS`** | 1 |
| **Endpoints per request** | 100 |
| **Test duration** | 300s |
| **Training batch size (executed in parallel)** | 1,000 entries every 500ms |
| **Model type** | XGBoost |

<br>

**Load Test Results**

| Metric | 1000 QPS | 1500 QPS | 2000 QPS |
| :--- | :--- | :--- | :--- |
| **Success rate** | 100% | 100% | 96.4% |
| **Total predictions** | 29.7M | 36.8M | N/A *(aborted)* |
| **Avg latency** | 17ms | 33ms | 300-480ms |
| **P50** | 16ms | 30ms | — |
| **P99** | 36ms | 81ms | — |
| **P999** | 81ms | 113ms | — |

**Comment by kaushikmitr** [comment]
@ahg-g closed https://github.com/kubernetes-sigs/gateway-api-inference-extension/pull/2318 check this
